addon-api/content-script/Tab.js

```diff
@@ -285,6 +285,9 @@ export default class Tab extends Listenable {
    * @returns {string} Hashed class names.
    */
   scratchClass(...args) {
+    if (!this._calledScratchClassReady)
```

Comment:
I think this would not catch cases where someone forgets to await the method call.

Reply:
This will only catch cases where a runAtComplete:false script forgot to await the method call and is using `addon.tab.scratchClass` before any other scripts from the same addon on the same tab that run on full complete start running.